utils/tracefs.c

```diff
@@ -1,67 +1,52 @@
 #include <fcntl.h>
+#include <linux/magic.h>
+#include <mntent.h>
 #include <stdio.h>
 #include <string.h>
+#include <sys/vfs.h>
 
 #include "utils/tracefs.h"
 #include "utils/utils.h"
 
-#define PROC_MOUNTINFO "/proc/self/mountinfo"
+#define PROC_MOUNTS_DIR_PATH "/proc/mounts"
+#define TRACEFS_DIR_PATH "/sys/kernel/tracing"
+#define OLD_TRACEFS_DIR_PATH "/sys/kernel/debug/tracing"
 
 static char *TRACING_DIR = NULL;
 
 static bool find_tracing_dir(void)
 {
 	FILE *fp;
-	char *line = NULL, fs_type[NAME_MAX], mount_point[PATH_MAX];
-	static char debugfs_suffix[] = "tracing";
-	bool debugfs_found = false;
-	size_t len;
+	struct mntent *ent;
+	struct statfs fs;
 
 	if (TRACING_DIR)
-		return false;
+		return true;
+
+	if ((!statfs(TRACEFS_DIR_PATH, &fs) && fs.f_type == TRACEFS_MAGIC) ||
+	    (!statfs(OLD_TRACEFS_DIR_PATH, &fs) && fs.f_type == TRACEFS_MAGIC)) {
+		xasprintf(&TRACING_DIR, "%s", TRACEFS_DIR_PATH);
+		return true;
+	}
```

Comment:
Hmm.. Does it really handle when only `/sys/kernel/debug/tracing` is available? This logic sets `TRACING_DIR` to `/sys/kernel/tracing` in both cases.

Reply:
I've incorporated feedback on what I missed.

Reply:
It's not that important but the expression "incorporated feedback" sounds awkward. You can just say that "I've updated it based on your feedback".